### PR TITLE
internal: migrate off nimbus

### PR DESCRIPTION
**Pull request by williaster**
#### :house: Internal

This PR migrates our development config / dependencies off of https://github.com/airbnb/nimbus which is no longer maintained and is more blackbox than we want.

Migrating off this will unblock things like https://github.com/airbnb/visx/pull/1578 and also allow us to upgrade beyond TS@3.8 which is the latest version `@airbnb/config-typescript` supports.

- [x] babel
  - [x] remove `@airbnb/config-babel` + add equivalent deps
  - [x] update scripts
    - [x] fix `build:workspaces` script / update documentation
  - [x] runs without error locally
  - [x] compiles both cjs + esm
  - [x] transforms typescript types to prop-types

- [x] eslint
  - [x] remove `@airbnb/config-eslint` + add equivalent deps
  - [x] update scripts
  - [x] upgrade lint configs
    - [x] fix breaking lint rules changes
  - [x] apply lint fixes
    - [x] `--fix`
    - [x] manual

- [x] typescript
  - [x] remove `@airbnb/config-typescript` + add equivalent deps
  - [x] update scripts
  - [x] commit `tsconfig*`s
  - [x] add `updateReferences` script
  - [x] types are generated / within-repo dependencies work correctly

- [x] prettier
  - [x] remove `@airbnb/config-prettier` + add equivalent deps
  - [x] update scripts
  - [x] config + ignore files are respected/found 
  - [x] runs in parallel
  - [x] fix deprecated `jsxBracketSameLine` option
  - [ ] add prettier to CI (separate PR)

- [x] jest
  - [x] remove `@airbnb/config-jest` + add equivalent deps
  - [x] update scripts
  - [x] implement setup files
  - [x] tests pass locally

- [x] remove `@airbnb/nimbus`
- [x] ci passes 🎉 🍹 
  - [x] update to `node>=16.10.0`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,12 +31,12 @@
...
     "babel": "yarn run babel:cjs && yarn run babel:esm",
-    "babel:cjs": "nimbus babel --clean --workspaces=\"@visx/!(demo)\"",
-    "babel:esm": "nimbus babel --clean --workspaces=\"@visx/!(demo)\" --esm",
+    "babel:cjs": "lerna exec --ignore @visx/demo --parallel -- babel --root-mode upward --delete-dir-on-start src/ --out-dir lib --extensions .ts,.tsx",
+    "babel:esm": "ESM=true lerna exec --ignore @visx/demo --parallel -- babel --root-mode upward --delete-dir-on-start src/ --out-dir esm --extensions .ts,.tsx",
     "build": "yarn run babel && yarn run type",
     "build:sizes": "yarn run ts ./scripts/computeBuildSizes.ts",
     "build:release": "yarn run ts ./scripts/performRelease/index.ts",
-    "build:workspaces": "nimbus babel --clean",
+    "build:workspaces": "yarn babel --delete-dir-on-start",
```
Comment (williaster) [inline package.json:39]: need to fix this one

File: package.json [hunk 2]
```diff
@@ -31,12 +31,12 @@
   "private": true,
   "scripts": {
     "babel": "yarn run babel:cjs && yarn run babel:esm",
-    "babel:cjs": "nimbus babel --clean --workspaces=\"@visx/!(demo)\"",
-    "babel:esm": "nimbus babel --clean --workspaces=\"@visx/!(demo)\" --esm",
+    "babel:cjs": "lerna exec --ignore @visx/demo --parallel -- babel --root-mode upward --delete-dir-on-start src/ --out-dir lib --extensions .ts,.tsx",
```
Comment (williaster) [inline package.json:34]: a big plus of `nimbus` was parallelization – prob have to use lerna for this

**Comment by JayWelsh** [comment]
Just want to say thank you so much for doing something about this!

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
+module.exports = {
+  root: true,
+  extends: [
+    // these are relics of nimbus, we could definitely simplify + consolidate
+    './config-eslint/base.js',
```
Comment (williaster) [inline .eslintrc.js:5]: this config is kind of out of control but I don't think trying to simplify it in this PR is a good idea given the amount of changes already here

**Review by ktmud — APPROVED** [review]
IIRC,  `prettier` should be able to fix all the missing newline at EOF errors, too?

**Comment by williaster** [comment]
Woot! Finally got CI to pass. Looked at the CI times and the TS task almost doubled because we lost some of the parallelization we got with nimbus. Everything else was mostly unchanged, though, and we can try to tune TS more in the future.

image

**Comment by JayWelsh** [comment]
Great work @williaster ! You've saved the day! 🥇 🥳

**Pull request merged**